File: backend/app/Controllers/Http/ServicesController.ts
import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
import Service from 'App/Models/Service'

export default class ServicesController {
  public async index({ request, response }: HttpContextContract) {
    const active = request.input('active', '')
    let query = Service.query()
    if (active) {
      query.where('is_active', active === '1')
    }
    const data = await query.orderBy('id', 'asc')
    return response.ok({ data })
  }

  public async store({ request, response }: HttpContextContract) {
    const data = request.only(['name', 'code', 'description', 'durationMinutes', 'price', 'capacity'])
    const service = await Service.create({
      ...data,
      durationMinutes: Number(data.durationMinutes || 30),
      price: Number(data.price || 0),
      capacity: Number(data.capacity || 1),
    })
    return response.created({ data: service, message: '服务创建成功' })
  }

  public async show({ params, response }: HttpContextContract) {
    const service = await Service.findOrFail(params.id)
    return response.ok({ data: service })
  }

  public async update({ params, request, response }: HttpContextContract) {
    const service = await Service.findOrFail(params.id)
    const data = request.only(['name', 'code', 'description', 'durationMinutes', 'price', 'capacity', 'isActive'])
    service.merge({
      ...data,
      durationMinutes: data.durationMinutes ? Number(data.durationMinutes) : undefined,
      price: data.price !== undefined ? Number(data.price) : undefined,
      capacity: data.capacity ? Number(data.capacity) : undefined,
    })
    await service.save()
    return response.ok({ data: service, message: '服务更新成功' })
  }

  public async destroy({ params, response }: HttpContextContract) {
    const service = await Service.findOrFail(params.id)
    await service.delete()
    return response.ok({ message: '服务已删除' })
  }
}

File: pet_boarding_scheduler/app/controllers/admin/services_controller.rb
module Admin
  class ServicesController < ApplicationController
    before_action :set_service, only: [:show, :edit, :update, :destroy]

    def index
      @services = Service.order(category: :asc, name: :asc)
      @services = @services.where(is_active: true) if params[:active_only]
    end

    def show; end

    def new
      @service = Service.new
    end

    def edit; end

    def create
      @service = Service.new(service_params)
      if @service.save
        redirect_to [:admin, @service], notice: "服务创建成功！"
      else
        render :new, status: :unprocessable_entity
      end
    end

    def update
      if @service.update(service_params)
        redirect_to [:admin, @service], notice: "服务更新成功！"
      else
        render :edit, status: :unprocessable_entity
      end
    end

    def destroy
      @service.destroy
      redirect_to admin_services_url, notice: "服务已删除。"
    end

    private

    def set_service
      @service = Service.find(params[:id])
    end

    def service_params
      params.require(:service).permit(
        :name, :description, :duration_minutes, :price,
        :category, :is_active
      )
    end
  end
end
